python/snewpy/models/extended.py

```diff
@@ -0,0 +1,110 @@
+        """Get neutrino luminosity from supernova cooling tail luminosity model.
+
+        Parameters
+        ----------
+        t : astropy.Quantity
+            Time to evaluate luminosity.
+        k : float
+            Power law factor (default: -1)
+        A : astropy.Quantity
+            Normalization factor (default: 1e51 erg/s)
+        tau_c : astropy.Quantity
+            Exponential decay characteristic timescale (default: 36 s)
+        alpha : float
+            Exponential decay factor (default: 2.66)
+
+        Returns
+        -------
+        astropy.Quantity
+            Luminosity calculated from cooling tail model.
+        """
+        if t.value < 0.5:
+            warn("Extended luminosity model not applicable to early times")
+        if A is None:
+            tf = self.t_final
+            Lf = self.L_final[flavor]
+            A = Lf/((tf.value**k) * np.exp(-1.*((tf/tau_c)**alpha)) * u.erg / u.s)
+        return A * (t.value**k) * np.exp(-1.*((t/tau_c)**alpha)) * u.erg / u.s
+
+    def extend(self, ts, k = -1., A = None, tau_c = 36. * u.s, alpha = 2.66):
+        """Extend supernova model to specific times.
+
+        Parameters
+        ----------
+        ts : astropy.Quantity
+            Times to add to supernova model.
+        k : float
+            Power law factor (default: -1)
+        A : astropy.Quantity
+            Normalization factor (default: None)
+        tau_c : astropy.Quantity
+            Exponential decay characteristic timescale (default: 36 s)
+        alpha : float
+            Exponential decay factor (default: 2.66)
+        """
+        for t in ts:
+            self.time = np.append(self.time, t)
+            for flavor in [Flavor.NU_E, Flavor.NU_X, Flavor.NU_E_BAR, Flavor.NU_X_BAR]:
+                L_ext = self.get_extended_luminosity(t, k = k, A = A, tau_c = tau_c, alpha = alpha, flavor = flavor)
+                self.luminosity[flavor] = np.append(self.luminosity[flavor], L_ext)
+                self.meanE[flavor] = np.append(self.meanE[flavor], self.meanE[flavor][-1])
+                self.pinch[flavor] = np.append(self.pinch[flavor], self.pinch[flavor][-1])
+
+    def get_integrated_luminosity(self, m_PNS, SRT, alpha, beta, gamma):
+        """Get integrated neutrino luminosity from shock revival time to 20 s.
+
+        Parameters
+        ----------
+        m_PNS : astropy.Quantity
+            Mass of the proto-neutron star.
+        SRT : astropy.Quantity
+            Shock revival time.
+        alpha : float
+            First cooling tail model parameter.
+        beta : float
+            Second cooling tail model parameter.
+        gamma : float
+            Third cooling tail model parameter.
+
+        Returns
+        -------
+        astropy.Quantity
+            Integrated luminosity calculated from Ekanger et al. (2023) model.
+        """
+        logE = (m_PNS.value * alpha) + (SRT.value * beta) + gamma
+        return (10**logE) * u.erg
```

Comment:
If I understand correctly, this fit is specific to the `Nakazato_2013` models (e.g. due to the 20s time range and the shock revival time) and does not make sense for other models? The function also does not appear to be used anywhere, so would it be better to delete it?